File: leetcode/1.go
package main

import (
	"fmt"
)

func twoSum(nums []int, target int) []int {
	length := len(nums)
	if length < 2{
		return nil
	}

	hashMap := make(map[int]int)
	for key,val := range nums {

		hashMapVal , ok := hashMap[target - val]
		if ok {
			return []int{hashMapVal, key}
		}
		hashMap[val] = key
	}

	return nil
}

func main() {
	sum := 22
	nums := []int{2, 7, 11, 15}
	fmt.Println(twoSum(nums, sum))
}

File: leetcode/1.php
<?php

// +----------------------------------------------------------------------------
// | nice版权所属 
// +----------------------------------------------------------------------------
// | Copyright (c) 2019 http://oneniceapp.com All rights reserved.
// +----------------------------------------------------------------------------
// | Author: dino.ma <dino_ma@163.com>
// +----------------------------------------------------------------------------
declare(strict_types=1);



class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {

        $length = count($nums);
        $map = [];
        for ($i = 0; $i < $length; $i++) {
            if (isset($map[$target - $nums[$i]])) {
                return [$i, $map[$target - $nums[$i]]];
            }
            $map[$nums[$i]] = $i;
        }


        return null;
    }
}



var_dump((new Solution())->twoSum([2, 7, 11, 15], 9));